### ISSUE TITLE
Map for WinUI

### ISSUE THREAD

**Issue by TonyLugg** [dotnet/maui#12694]
### Description

The absence of a WinUI from MAUI Maps is problematic.  I recognize that MS does not have support for its Bing Maps in WinUI, which is the issue for MAUI.  However, MS needs to understand that this absence prevents us from properly supporting Windows in our MAUI app.

### Public API Changes

Add WinUI Bing Maps support for Microsoft.Maui.Controls.Maps.

### Intended Use-Case

Need the map for WinUI.

**Comment by mattleibow** [comment]
This is frustrating. 

Just linking some info for future peoples.

 - This is the reason: https://github.com/dotnet/maui/issues/11169#issuecomment-1306893533
 - WinUI issue: https://github.com/microsoft/microsoft-ui-xaml/issues/3024
 - Proposal for a workaround: https://github.com/CommunityToolkit/Maui/issues/605
 - Draft for that workaround: https://github.com/CommunityToolkit/Maui/pull/604

I do know the WinUI team is looking at issues these days, so please leave your vote on https://github.com/microsoft/microsoft-ui-xaml/issues/3024

**Comment by Mogikan** [comment]
> We've moved this issue to the Backlog milestone. This means that it is not going to be worked on for the coming release. We will reassess the backlog following the current release and consider this item at that time. To learn more about our issue management process and to have better expectation regarding different types of issues you can read our [Triage Process](https://github.com/dotnet/maui/blob/main/docs/TriageProcess.md).

Why was it removed from backlog? It was there before win ui 3 release 2 years ago and then it suddenly disappeared.

**Comment by TonyLugg** [comment]
Of course, everything gets moved to the backlog.  It means it won't be done.  I may stop giving feedback altogether, it's a waste of time.

**Comment by TFreudi1** [comment]
Very frusting ! I have some map based Xamarin-Apps that run on Android/Windows/iOS and allthough I want, I can't move them to .NET MAUI due to the absence of a Windows Map control. All alternative products that I tested have license issues, or issues with bing maps, or don't show bing maps at all. And even if they would work it means that the #if WINDOWS preprocessor commands are back :-(

**Comment by robloo** [comment]
Use https://github.com/Mapsui/Mapsui. It's probably the best all-around map control left and it's now MIT and supports Maui in the upcoming version. Actually, XF was the best supported platform so I expect Maui support to be pretty good. I don't use Maui on my end but I just wanted to let you all know of this control.

**Comment by mattleibow** [comment]
Please leave a comment on https://github.com/microsoft/microsoft-ui-xaml/issues/3024 so the WinUI team can correctly prioritize features. We cannot implement this feature in Maui as the OS does not have maps for us to access.

**Comment by TonyLugg** [comment]
@robloo Thanks for the link.  It looks like Mapsui uses OpenStreetMaps as its data source, which we found to be somewhat unreliable in certain areas.  A pretty good free option though.

**Comment by robloo** [comment]
You can use whatever mapping provider you want. It is configurable. Bing maps was used on my end and it's fully supported. You just need the key/token same as other map controls.

**Comment by TonyLugg** [comment]
👍

**Comment by TFreudi1** [comment]
MapSui not really work , I have 2 Pages and only the first time it shows a map on the map page. Itried a lot of things and opened an issue there.
[https://github.com/Mapsui/Mapsui/issues/1896](url)

**Comment by TFreudi1** [comment]
Looks like Mapsui is not a solution if you have multiple pages. I only get it to work with a complete new initialisation every time you slide the mappage in, this is not usable.

**Comment by robloo** [comment]
It certainly has issues but it's better than nothing and works in most cases. The project, while slow, accepts PRs. You can also fork the code.

**Comment by andrasjani** [comment]
still need it 🙏

**Comment by TFreudi1** [comment]
Any progress on that ?

**Comment by andrasjani** [comment]
> Any progress on that ?

There is a temporary solution in the toolkit which is based on webview. More info here:
https://github.com/CommunityToolkit/Maui/releases/tag/1.0.0-maps

**Comment by jfversluis** [comment]
Looked into this for a little, but basically all the functionality that is implemented in .NET MAUI maps right now is missing for the WinUI version: https://github.com/microsoft/microsoft-ui-xaml/issues/3024#issuecomment-1932950478

So this is not really feasible to implement atm

**Comment by malsabi** [comment]
@jfversluis Any updates on the current status of WinUI Map or any plans on implementing in .NET MAUI?

**Comment by jfversluis** [comment]
> @jfversluis Any updates on the current status of WinUI Map or any plans on implementing in .NET MAUI?

As far as I know not much has changed unfortunately

**Comment by bmacombe** [comment]
This is a commercial product, but it is a very capable mapping platform. I use it on WinUI, Android and iOS and it works great.

https://thinkgeo.com/mobile-maps

**Comment by robloo** [comment]
WinUI added a map control in WinAppSdk 1.5 -- it's a wrapper around a webview and has issues -- but it should be usable enough for MAUI and I'm pretty sure that's why they added it. Are the teams not talking together on this anymore?

https://github.com/microsoft/microsoft-ui-xaml/issues/3024

---

For app developers you are going to want to use 3rd party controls though. Mapsui is a free alternative.

**Comment by jfversluis** [comment]
Implemented through https://github.com/dotnet/maui/pull/34138, however it will not be up to par with Android and iOS. This is due to limitations on the WinUI MapControl. If you would like to see more implemented, you will have to turn to them for more support. Still hope this will be helpful for people!

It will land in .NET MAUI 10 SR6
